src/dash/application/content-types/content-types-repository.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+// phpcs:disable Yoast.NamingConventions.NamespaceName.TooLong
+namespace Yoast\WP\SEO\Dash\Application\Content_Types;
+
+use Yoast\WP\SEO\Dash\Application\Taxonomies\Taxonomies_Repository;
+use Yoast\WP\SEO\Dash\Domain\Content_Types\Content_Type;
+use Yoast\WP\SEO\Dash\Domain\Content_Types\Content_Types_List;
+use Yoast\WP\SEO\Helpers\Post_Type_Helper;
+
+/**
+ * The repository to get content types.
+ */
+class Content_Types_Repository {
+
+	/**
+	 * The post type helper.
+	 *
+	 * @var Post_Type_Helper
+	 */
+	protected $post_type_helper;
+
+	/**
+	 * The content types list.
+	 *
+	 * @var Content_Types_List
+	 */
+	protected $content_types_list;
+
+	/**
+	 * The taxonomies repository.
+	 *
+	 * @var Taxonomies_Repository
+	 */
+	private $taxonomies_repository;
+
+	/**
+	 * The constructor.
+	 *
+	 * @param Post_Type_Helper      $post_type_helper      The post type helper.
+	 * @param Content_Types_List    $content_types_list    The content types list.
+	 * @param Taxonomies_Repository $taxonomies_repository The taxonomies repository.
+	 */
+	public function __construct(
+		Post_Type_Helper $post_type_helper,
+		Content_Types_List $content_types_list,
+		Taxonomies_Repository $taxonomies_repository
+	) {
+		$this->post_type_helper      = $post_type_helper;
+		$this->content_types_list    = $content_types_list;
+		$this->taxonomies_repository = $taxonomies_repository;
+	}
+
+	/**
+	 * Returns the content types array.
+	 *
+	 * @return array<array<string,array<string, array<string, array<string, string|null>>>>> The content types array.
```

Comment:
I get the point, but I think the main reason we're doing this is to later hook a static analysis tool like PHPStan which will depend on those types. So that wouldn't work in this case, so I'm proposing we leave it as is :)